internal/transport/http2_client.go

```diff
@@ -1147,7 +1151,7 @@ func (t *http2Client) handleGoAway(f *http2.GoAwayFrame) {
 	id := f.LastStreamID
 	if id > 0 && id%2 != 1 {
```

Comment:
The `>0` here is unnecessary; please remove it.

Reply:
Actually this turned out to be needed (since a stream id of zero will be sent in case of goaway on an RPC-less connection, and this isn't an error AFAICT). My error message was previously incorrect, since it's actually a protocol error to receive <i>even-numbered</i> non-zero stream IDs in goaways

Reply:
Oh wow.. I misunderstood the logic, too (possibly because of your comment).

Maybe rewrite this to: `if id > 0 && id%2 == 0` ?